src/main/java/net/fabricmc/loader/impl/util/mappings/TinyRemapperMappingsHelper.java

```diff
@@ -29,19 +26,28 @@ private static IMappingProvider.Member memberOf(String className, String memberN
 		return new IMappingProvider.Member(className, memberName, descriptor);
 	}
 
-	public static IMappingProvider create(TinyTree mappings, String from, String to) {
+	public static IMappingProvider create(MappingTree mappings, String from, String to) {
 		return (acceptor) -> {
-			for (ClassDef classDef : mappings.getClasses()) {
-				String className = classDef.getName(from);
-				acceptor.acceptClass(className, classDef.getName(to));
+			final int fromId = mappings.getNamespaceId(from);
+			final int toId = mappings.getNamespaceId(to);
 
-				for (FieldDef field : classDef.getFields()) {
-					acceptor.acceptField(memberOf(className, field.getName(from), field.getDescriptor(from)), field.getName(to));
+			for (MappingTree.ClassMapping classDef : mappings.getClasses()) {
+				final String className = classDef.getName(fromId);
+				String dstName = classDef.getName(toId);
+
+				if (dstName == null) {
+					dstName = className;
+				}
+
+				acceptor.acceptClass(className, dstName);
+
+				for (MappingTree.FieldMapping field : classDef.getFields()) {
+					acceptor.acceptField(memberOf(className, field.getName(fromId), field.getDesc(fromId)), field.getName(toId));
```

Comment:
Will field.getName(toId) return null and break stuff? Same question for method.getName(toId) below.

Reply:
This matches the code used in loom, its possible it could do just not sure if it will in practice?